data/json/monsters.json

```diff
@@ -3070,13 +3070,34 @@
         {
             "type" : "bite",
             "cooldown" : 10,
-            "damage_max_instance" : {
-                "type" : "stab",
-                "amount" : 10,
-                "armor_multiplier" : 0.2
-            }
+            "damage_max_instance" : [
+                { "damage_type" : "stab", "amount" : 10, "armor_multiplier" : 0.2 }
+            ]
         },
-        ["IMPALE", 25]
```

Comment:
Yes, that would be a good solution. Inline definition should remain possible and be encouraged for small definitions.

This also needs some documentation for those creating mods, perhaps in a similar format to the gun modes documentation I wrote (as a series of worked examples).

Also needs to update the linter ruleset. Given your proposed syntax is relatively complex (and I don't see how you can avoid that) it will be a good test that the linter rules are expressive enough. There should be enough examples in the documentation and/or existing ruleset but let me know if you have any problems with that.